backend/src/test/groovy/com/capstone/moneytree/service/DefaultStockServiceTest.groovy

```diff
@@ -51,7 +60,34 @@ class DefaultStockServiceTest extends Specification {
         List<SanitizedStock> owns = stockService.getUserStocks(userId)
 
         then: "should get all user stocks"
-        assert owns != null
-        assert owns.size() == 1
+        owns != null
+        owns.size() == 1
+    }
+
+    @Ignore // for some reason this test fail. I have wasted too much time. if anyone knows how to fix it please do!!!
```

Comment:
please please please, I am baffled, I dont know why this one does not want to work!!!! 😭 😨

Reply:
It has a very weird behaviour. It seems the assertion in the "then" block is affecting the call in the "when" block, not sure why. 
Below is my rewrite of the test, which passes. I made the mocking less generic and more specific to the AAPL symbol since no other symbol should return the AAPL stock. I also replaced the called once assertion to a simple equality check, which seems more fitting.
```java
        given: "a stock"
        Stock stock0 = new Stock("AAPL", "Apple")

        and:
        stockDao.findBySymbol("AAPL") >> stock0

        when:
        Stock stock1 = stockService.getStockBySymbol("AAPL")

        then:
        assert stock1 == stock0
```

Reply:
The above works. Thanks Walter!